dxr/config.py

```diff
@@ -105,7 +105,7 @@ def __init__(self, input, relative_to=None):
                     AbsPath,
                 Optional('workers', default=if_raises(NotImplementedError,
                                                       cpu_count,
-                                                      1)):
+                                                      1) / 2):
```

Comment:
I'm going to move this to a separate PR in the interests of landing the bug fix quickly.

I don't think it is worth making another dependency for. It's only a default and I think we can come up with a good heuristic. My VM is also set to have cores = physical + logical, but if the OS then tries to really use that many threads, it really seizes up the machine. It might be worth experimenting with the Clang plugin to find out what the best number of cores is, rather than trying to guess. For Rust going from 8 to 4 workers meant 90-ish to 50-ish minutes, which is a massive saving. Lets chat about this later and I'll send in a PR when we figure out what to do.